layer-atlas/src/main/java/com/layer/atlas/messagetypes/location/LocationCellFactory.java

```diff
@@ -147,13 +146,19 @@ public void onClick(View v) {
         Location location = (Location) v.getTag();
         String encodedLabel = (location.mLabel == null) ? URLEncoder.encode("Shared Marker") : URLEncoder.encode(location.mLabel);
         Intent intent;
+        Context context = v.getContext();
         if ("baidu".equals(mMapType)) {
             intent = new Intent();
             intent.setData(Uri.parse("baidumap://map/marker?location=" + location.mLatitude + "," + location.mLongitude + "&title=Pinned Point"));
-        } else {
-            intent = new Intent(Intent.ACTION_VIEW, Uri.parse("geo:0,0?q=" + location.mLatitude + "," + location.mLongitude + "(" + encodedLabel + ")&z=16"));
+            try {
+                context.startActivity(intent);
+                return;
+            } catch (Exception e) {
```

Comment:
Don't catch `Exception`
catch `ActivityNotFoundException`, and show a toast saying please install baidu map.